source/elements/oneTBB/source/algorithms/blocked_ranges/blocked_rangeNd_cls.rst

```diff
@@ -0,0 +1,151 @@
+.. SPDX-FileCopyrightText: 2019-2024 Intel Corporation
+..
+.. SPDX-License-Identifier: CC-BY-4.0
+
+===============
+blocked_rangeNd
+===============
+**[algorithms.blocked_range3d]**
+
+Class template that represents a recursively divisible N-dimensional half-open interval.
+
+A ``blocked_rangeNd`` is the N-dimensional extension of ``blocked_range``, ``blocked_range2d`` and ``blocked_range3d``.
+We can interpret it as a tensor-product of N instances of ``blocked_range``.
+Different to ``blocked_range2d`` and ``blocked_range3d``, all ranges have to be specified over the same type Value.
+Therefore, blocked_rangeNd<int,2> is a specialisation of blocked_range2d where both dimensions have int type.
+
+.. code:: cpp
+
+    namespace oneapi {
+        namespace tbb {
+            template<typename Value, unsigned int N>
+            class blocked_rangeNd {
+            public:
+                // Types
+                using value_type = Value;
+                using dim_range_type = tbb::blocked_range<Value>;
+
+                // Constructors
+                blocked_rangeNd(const dim_range_type<Is>&... args);
+                blocked_rangeNd(
+                  Value begin[N],
+                  Value end[N],
+                  typename tbb::blocked_range<value_type>::size_type grainsize
+                );
+                blocked_rangeNd(blocked_rangeNd& r, proportional_split proportion); 
+                blocked_rangeNd(blocked_rangeNd& r, split proportion); 
+               
+                // Capacity
+                bool empty() const;
+
+                // Access
+                bool is_divisible() const;
+                static constexpr unsigned int ndims();
```

Comment:
This static member function seems not described anywhere.

Also I would think of a better name; this one seems too cryptic. Maybe `dim_count` or even `dimension_count`?

Reply:
I will go with `dim_count`, as `dim` is already used in place of `dimension` in several other instances in the context of blocked_rangeNd.